### ISSUE TITLE
Report generated with wrong stats when using JBehave GivenStories

### ISSUE THREAD

**Issue by wilker7ribeiro** [serenity-bdd/serenity-core#631]
After executing all tests, my html report (target/site/serenity/index.html) counts my tests incorrectly.

The problem is that Serenity is discarting the examples from scenarios that use GivenStories, and counting examples from the scenary of the GivenStory story.

My Stories are something like that:
___
Login
**Narrative:**
**In order to** .... log-in
**As A** User
**I want** .... log-in

**Scenary**: Login - Successful
**Meta**: @id1 scenary1
**Given** the user be at login page.
**When** insert `<login>`
**When** insert `<password>`
**When** click the button log-in
**Then** the site should navigate to home page
**Examples**:
|login  |password|
|login1|Test123|
___
Edit Profile
Login
**Narrative:**
**In order to** .... edit my profile
**As A** User
**I want** .... edit

**Scenary**: Edit Profile - Field Required
**Meta**: @id1 scenary1
**GivenStories**: path/to/my/login.story
**Given** the user be at home page.
**When** click the button edit profile
**When** insert `<Name>`
**When** insert `<Last Name>` 
**When** insert `<Age>`
**When** insert `<CPF>`
**Then** the site should show `<message>`
Examples:
|Name|Last Name|Age|CPF                   |message                 |
|------|Smith       |35  |072.651.495-05|* Name field is required |
|John  |----------|35  |072.651.495-05|* Last Name field is required |
|John  |Smith       |----|072.651.495-05|* Age field is required    |
|John  |Smith       |35  |-----------------|* CPF field is required    |
___
My report:

![serenityerro2](https://cloud.githubusercontent.com/assets/21000731/21273253/cdf8f14c-c3a9-11e6-8f38-1520c9263733.png)

Scenary Edit Profile with wrong results:
![serenityerro](https://cloud.githubusercontent.com/assets/21000731/21273177/82758a64-c3a9-11e6-9112-a69f39289ce3.png)

The example table above must be the Edit Profile scenary.
I can reproduce this bug with larger tests
If the GivenStory scenary doesn't have example table, this problem doesn't occur.

**Comment by wakaleo** [comment]
I'll take a look. Using an example table with a single row isn't really the intended use of example tables, though.

**Comment by wilker7ribeiro** [comment]
Thank you for the fast response!

This example that I sent you focuses in this specific bug. In my real project, I have a lot more Storys, Scenarios and Steps.
Example:
___
Login
**Narrative**:
**In order to** .... log-in
**As A** User
**I want** .... log-in

**Scenary**: Login - Successful
**Meta**: @id1 scenary1
**Given** the user be at login page.
**When** insert `<login>`
**When** insert `<password>`
**When** click the button log-in
**Then** the site should navigate to home page
**Examples**:
|login |password|
|login1|Test123|

**Scenary**: Login - Failed
**Meta**: @id2 scenary2
**Given** the user be at login page.
**When** insert `<login>`
**When** insert `<password>`
**When** click the button log-in
**Then** the site should show 'Wrong user or password'
**Examples**:
|login |password|
|123456|Abc321|
|AAAbbb|asdfgh|
|zxcvbnm|qwerty|
___

This way, all my scenarios use the same methods and I don't need to rewrite them:

```
@Given("the user be at login page")
public void givenTheUserBeAtLoginPage() {
	steps.verifyLoginPage();
}
	
@When("insert <login>")
public void whenInsertLogin(@Named("login") String login) {
	steps.insertLogin(login);
}
	
@When("insert <password>")
public void whenInsertPassword(@Named("password") String password) {
	steps.insertPassword(password);
}
	
@When("click the button log-in")
public void whenClickTheButtonLogin() {
	steps.clickButtonLogin();
}
	
@Then("the site should show '$message'")
public void thenTheSiteShouldShowWrongUserOrPassword(String message) {
	steps.verifyMessage(message);
}

@Then("the site should navigate to home page")
public void thenTheSiteShouldNavigateToHomePage() {
	steps.verifyHomePage();
}
```
If I change my Given, When and Then and doesn't use 1 row example table, I should rewrite and duplicate the Methods, one for scenarios that uses example table, other for those who don't.

**Comment by wilker7ribeiro** [comment]
Also, seems like browser is being closed after scenarios even if 'restart.browser.each.scenario' property is set to false and 'serenity.use.unique.browse' to true. It screws up the GivenStories thing.
After I login (GivenStory) my browser closes and my test fails at "Given the user be at home page."
I noticed that using v1.42 or previuos versions this doesn't happen.

**Comment by wakaleo** [comment]
Could you try with 1.2.2-rc.1 (serenity-core) 1.17.0 (serenity-jbehave)?

**Comment by wilker7ribeiro** [comment]
That's exactly what I did.
I was using v1.1.36 serenity with v1.13.0 serenity-jbehave and everything worked fine, except wrong reports.
Then I updated to 1.2.2-rc.1 with 1.17.0 and the browser started to close after each scenario.
I checked every version in maven central and it seems to start after v1.1.42 serenity-core.
Let me know if you can't reproduce so I know that is something with my project.

**Comment by wakaleo** [comment]
Are the reports correct for this version?

**Comment by wilker7ribeiro** [comment]
Same thing still, taking examples from GivenStories dataTable and test numbers incorrects.
But now all my tests with GivenStories failed (chrome closes and reopen with 'data:,' blank).

**Comment by wakaleo** [comment]
Could you provide a sample project that reproduces the issue?

**Comment by wilker7ribeiro** [comment]
For report issue:
[jbehave-webtests.zip](https://github.com/serenity-bdd/serenity-core/files/663753/jbehave-webtests.zip)

Working in this sample to reproduce the browser issue...

**Comment by wilker7ribeiro** [comment]
Here a sample that reproduce both issues, browser closing and wrong reports:
[jbehave-webtests-browserbug.zip](https://github.com/serenity-bdd/serenity-core/files/664168/jbehave-webtests-browserbug.zip)

**Comment by wilker7ribeiro** [comment]
All these issues are fixed with serenity-jbehave v1.18.0 and serenity v1.2.2-rc.4 :)
The only thing that still wrong is the Scenario Outline:

![bug1](https://cloud.githubusercontent.com/assets/21000731/21452951/bea6e094-c8f1-11e6-997d-626980f5a131.jpg)

![bug2](https://cloud.githubusercontent.com/assets/21000731/21452993/209b1806-c8f2-11e6-88c9-0a45004e13ad.jpg)

**Comment by wakaleo** [comment]
The scenario outline issue should be fixed in the next release.

**Comment by wakaleo** [comment]
Fixed in 1.19.0
